fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java

```diff
@@ -807,6 +830,19 @@ public Expression visitDate_sub(Date_subContext ctx) {
                 + ", supported time unit: YEAR/MONTH/DAY/HOUR/MINUTE/SECOND", ctx);
     }
 
+    @Override
+    public Expression visitDoublePipes(DorisParser.DoublePipesContext ctx) {
+        return ParserUtils.withOrigin(ctx, () -> {
+            Expression left = getExpression(ctx.left);
+            Expression right = getExpression(ctx.right);
+            if (sessionVariable.getSqlMode() == SqlModeHelper.MODE_PIPES_AS_CONCAT) {
```

Comment:
Could we use `ConnectContext.get().getSessionVaraible().getSqlMode()` ?

Reply:
done